### PR TITLE
[CP-3004] Refactored button action to accept multiple actions

### PR DESCRIPTION
**Pull request by mkurczewski**
JIRA Reference: [CP-3004]

### :memo: Description ️

Refactored button's `action` property to accept an array of actions that can be executed one by one at single click.

### :muscle: Checklist before requesting a review - nice to have

- getState function in async thunk actions is correctly typed
- redux selectors are used in components / prop drilling is reduce

### :exclamation: Checklist before merging a pull request

- change went through the QA process
- translations are updated in dedicated application
- [CHANGELOG.md](./CHANGELOG.md) is updated

[CP-3004]: https://appnroll.atlassian.net/browse/CP-3004?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/generic-view/feature/src/lib/setup-component.tsx [hunk 1]
```diff
@@ -79,17 +79,15 @@ export const setupComponent = <P extends object>(
...
 
-    const editableProps = {
-      ...dataProps,
-      config: {
-        ...config,
-      },
-      data: undefined as unknown,
-    }
+    const editableProps = JSON.parse(
+      JSON.stringify({
```
Comment (dkarski) [inline libs/generic-view/feature/src/lib/setup-component.tsx:83]: Hey! I noticed you're using `JSON.stringify/parse` here — are you aiming for a deep copy? If so, maybe `_.cloneDeep()` would be clearer and more predictable in terms of intent. What do you think?

File: libs/generic-view/feature/src/lib/setup-component.tsx [hunk 2]
```diff
@@ -163,6 +169,38 @@ export const setupComponent = <P extends object>(
   }
 }
 
+const processFormFields = (
+  field: DataProviderExtendedField,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  value: any
```
Comment (dkarski) [inline libs/generic-view/feature/src/lib/setup-component.tsx:175]: Maybe replace `any` with `unknown` and use union type checks (e.g., `is List`, `is Boolean`) for safer type checking?

**Review by dkarski — COMMENTED** [review]
The change looks good to me. I just see two small things to address.

Please also ensure that it doesn’t affect the configuration coming from Kompakt or the existing e2e tests. If it does, I suggest making the necessary changes in Kompakt and updating the e2e tests as part of this PR before merging.

File: libs/generic-view/feature/src/lib/setup-component.tsx (hunk 1 above)
Comment (mkurczewski) [reply libs/generic-view/feature/src/lib/setup-component.tsx]: Good point. Changed in [34c1131](https://github.com/mudita/mudita-center/pull/2076/commits/34c113113a5c96cbee59a520d9a1f43401af6e57)

File: libs/generic-view/feature/src/lib/setup-component.tsx (hunk 2 above)
Comment (mkurczewski) [reply libs/generic-view/feature/src/lib/setup-component.tsx]: Fixed in [`c8028ba` (#2076)](https://github.com/mudita/mudita-center/pull/2076/commits/c8028ba6b8efa5786ed2b360909d7f6c983e4cea)

**Review by dkarski — APPROVED** [review]

**Pull request merged**